File: src/main/java/xyz/oribuin/lilori/commands/music/CmdPause.java
package xyz.oribuin.lilori.commands.music;

import com.sedmelluq.discord.lavaplayer.player.AudioPlayer;
import net.dv8tion.jda.api.EmbedBuilder;
import xyz.oribuin.lilori.managers.command.Command;
import xyz.oribuin.lilori.managers.command.CommandEvent;
import xyz.oribuin.lilori.managers.music.AudioPlayerSendHandler;
import xyz.oribuin.lilori.managers.music.TrackManager;

import java.awt.*;
import java.util.Collections;

public class CmdPause extends Command {

    public CmdPause() {
        this.name = "Pause";
        this.aliases = Collections.emptyList();
        this.description = "Pause the music";
    }

    @Override
    public void executeCommand(CommandEvent event) {
        TrackManager tm = TrackManager.getInstance(event.getGuild());

        if (event.getMember().getVoiceState() == null || !event.getMember().getVoiceState().inVoiceChannel()) {
            event.reply(event.getAuthor().getAsMention() + ", Could not change volume since you are not in the voice channel");
            return;
        }

        AudioPlayer player = tm.getMusicManager().player;

        String msg;

        if (player.isPaused()) {
            player.setPaused(false);
            msg = "Playback is no longer paused";
        } else {
            player.setPaused(true);
            msg = "Playback is now paused";
        }

        EmbedBuilder embedBuilder = new EmbedBuilder()
                .setAuthor("\uD83C\uDFB5 " + msg)
                .setColor(Color.RED)
                .setDescription("Type " + event.getPrefix() + "pause to pause/unpause!")
                .setFooter("Created by Oribuin", "https://imgur.com/ssJcsZg.png");

        event.getTextChannel().sendMessage(event.getAuthor().getAsMention()).embed(embedBuilder.build()).queue();
    }
}

File: src/main/kotlin/xyz/oribuin/lilori/commands/music/CmdPause.kt
package xyz.oribuin.lilori.commands.music

import net.dv8tion.jda.api.EmbedBuilder
import xyz.oribuin.lilori.LilOri
import xyz.oribuin.lilori.handler.Category
import xyz.oribuin.lilori.handler.Command
import xyz.oribuin.lilori.handler.CommandEvent
import xyz.oribuin.lilori.managers.music.TrackManager.Companion.getInstance
import java.awt.Color

class CmdPause(bot: LilOri) : Command(bot) {
    init {
        name = "Pause"
        category = Category(Category.Type.MUSIC)
        aliases = emptyList()
        description = "Pause the music"
        arguments = emptyList()
    }

    override fun executeCommand(event: CommandEvent) {
        val tm = getInstance(event.guild)?: return

        // Check if the user is in a voice channel
        if (event.member.voiceState == null || event.member.voiceState?.inVoiceChannel() == false) {
            event.sendEmbedReply("❗ Unable to pause", "You cannot toggle the pausing because you are not in a voice channel!")
            return
        }

        // Define the player
        val player = tm.musicManager.player

        // Toggle pausing
        if (player.isPaused) {
            player.isPaused = false
            event.sendEmbedReply("\uD83C\uDFA7 Paused Music", "You have paused the music!")
        } else {
            player.isPaused = true
            event.sendEmbedReply("\uD83C\uDFA7 Unpaused Music", "You have unpaused the music!")
        }
    }
}